addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -281,12 +331,21 @@ export const editorCommands = {
                     }
                 }
             }
+            // When multiple <p>text1</p><p>text2</p><p>text3</p><p>text3</p> only start
+            // and end were converting to <li>, and rest were pasted as <p> only, convert
+            // remaining <p> to <li> before pasting on list.
+            if (isBlock(nodeToInsert) && currentNode.nodeName === 'LI') {
+                setTagName(nodeToInsert, 'LI');
+            }
             if (insertBefore) {
                 currentNode.before(nodeToInsert);
                 insertBefore = false;
             } else {
                 currentNode.after(nodeToInsert);
             }
+            if (nodeToInsert.nodeName === 'LI' && nodeToInsert.classList.contains('oe-nested')) {
+                convertedList = convertList(nodeToInsert.firstChild);
```

Comment:
How come we're doing this again? I thought `convertList` was already converting its nested descendants recursively?

Reply:
As you said it was converting its nested **descendants** recursively not it adjacent siblings
for let's say
```html
<ul>
    <li>a</li>
    <li class="oe-nested">
        <ul>
            <li>b</li>
            <li>c</li>
        </ul>
    </li>
    <li>d</li>
</ul>
```
if I paste that in a `ol` with this code the node which are not `firstElementChild` or `lastElementChild` are converted.